src/pre_defined_classes/mathematical_program/hard/q8/MainTest.java

```diff
@@ -0,0 +1,29 @@
+package pre_defined_classes.mathematical_program.hard.q8;
+
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    //Parsons
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("The 7th term in the geometric sequence is: "),
+                new DoubleLiteral("math"),
+                new StringLiteral("E-4")
```

Comment:
Can you not just catch the decimal value here?

Reply:
As stated in a previous comment, that I closed after writing it this way, the doubleLiteral would not accept "E-4", the only way to use the DoubleLiteral at all was if I detached the E-4 from it, because E is not a number and DoubleLiteral therefore can't accept it

Reply:
This is a framework problem for @aadiraju and I to resolve